tlatools/org.lamport.tlatools/src/util/SimpleFilenameToStream.java

```diff
@@ -158,9 +173,8 @@ public String getFullPath() {
    *
    *  @param  module name, used as basis of path name to the file that should contain it
    */
-  private final File locate(String name)
```

Comment:
Why is changing `locate` necessary?  This creates a non-negligible risk.

Reply:
Instead of introducing the `LocateLibraryPathContainer`, can't the library path be added as a new field to `TLAFile`?

Reply:
Great idea! As we return `File` here, I've thought that it was not possible to use TLAFile, but I've done as you said and now we don't need `getLibraryPath` for the `FilenameToStream` interface, much more elegant o/